### PR TITLE
Automated JS minification with google closure

### PR DESCRIPTION
**Pull request by slowr**
### Description of PR

Added google closure compiler step when building frontend container.

What component(s) does this PR affect?

- [ ] Back-End (Database, Microservices, Containers, etc)
- [x] Front-End (Flask, API, etc)
- [ ] Docs
- [x] Build System

Does the PR require changes on other components? If yes, please mark the components:

- [ ] Back-End (Database, Microservices, Containers, etc)
- [ ] Front-End (Flask, API, etc)
- [ ] Docs
- [ ] Build System

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Docs update
- [ ] None of the above

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [x] This change requires a change in the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,23 @@
-FROM jfloff/alpine-python:3.6
-
+FROM femtopixel/google-closure-compiler:0.1.5 as builder
```
Comment (vkotronis) [inline frontend/Dockerfile]: Does this include also the Python functionality we need?

File: frontend/Dockerfile [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 WORKDIR /root
 
 COPY . ./
+RUN rm /root/webapp/static/js/custom/*
+COPY --from=builder /root/prod/ /root/webapp/static/js/custom/
 
 RUN apk update && apk add --no-cache openssl-dev libffi-dev py-openssl sqlite-dev sqlite
 RUN pip install --upgrade 'pip<19.0'
 RUN pip --no-cache-dir install -r requirements.txt
-RUN pip --no-cache-dir install gunicorn
```
Comment (vkotronis) [inline frontend/Dockerfile]: Why is gunicorn deleted from here? It is installed only as a Python package from now on?

**Comment by vkotronis** [comment]
Also please do not forget to update the CHANGELOG

File: frontend/Dockerfile (hunk 1 above)
Comment (slowr) [reply frontend/Dockerfile]: we use two images if you see. this one is only for minifying the JS

File: frontend/Dockerfile (hunk 2 above)
Comment (slowr) [reply frontend/Dockerfile]: exactly the same.. this one was installing the latest version

File: frontend/Dockerfile (hunk 1 above)
Comment (vkotronis) [reply frontend/Dockerfile]: Where is the other image jfloff/alpine-python:3.6? I think I am missing what will happen with merge. Will we have two frontend images, and 2 Dockerfiles?

**Pull request merged**